src/lightning/pytorch/strategies/xla.py

```diff
@@ -201,13 +201,10 @@ def reduce(
 
     def setup_distributed(self) -> None:
         self._launched = True
-        self.set_world_ranks()
         rank_zero_only.rank = self.global_rank
 
     def set_world_ranks(self) -> None:
-        if self.cluster_environment is None:
-            return
-        rank_zero_only.rank = self.cluster_environment.global_rank()
+        pass
```

Comment:
Avoid calling anything that would initialize the computation client in the main process.

Reply:
Landing this separately in https://github.com/Lightning-AI/lightning/pull/16966